lib/pc_patch.c

```diff
@@ -608,3 +608,56 @@ PCPOINT *pc_patch_pointn(const PCPATCH *patch, int n)
 	pcerror("%s: unsupported compression %d requested", __func__, patch->type);
 	return NULL;
 }
+
+/** set schema for patch */
+PCPATCH*
+pc_patch_set_schema(PCPATCH *patch, const PCSCHEMA *new_schema, double defaultvalue)
+{
+	PCDIMENSION** new_dimensions = new_schema->dims;
+	PCDIMENSION* old_dimensions[new_schema->ndims];
+	double default_values[new_schema->ndims];
+	PCPATCH_UNCOMPRESSED *paout = NULL;
+	PCPOINTLIST *opl, *npl;
+	PCPOINT *opt, *npt;
+	size_t i, j;
+	double val;
+
+	for ( j = 0; j < new_schema->ndims; j++ )
+	{
+		PCDIMENSION *ndim = new_dimensions[j];
+		old_dimensions[j] = pc_schema_get_dimension_by_name(
+				patch->schema, ndim->name);
+		default_values[j] = pc_value_scale_offset(defaultvalue, ndim);
```

Comment:
I agree. A user wanting to set the raw value would have to use the forthcoming pcpoint version.